keras/preprocessing/sequence.py

```diff
@@ -216,3 +218,95 @@ def _remove_long_seq(maxlen, seq, label):
             new_seq.append(x)
             new_label.append(y)
     return new_seq, new_label
+
+
+class TimeseriesGenerator(Sequence):
+    """Utility class for generating batches of temporal data.
+
+    This class takes in a sequence of data-points gathered at 
+    equal intervals, along with time series parameters such as 
+    stride, delay etc., to produce batches for training/validation.
+
+    # Arguments
+        data: indexable generator (such as list or Numpy array)
+            containing consecutive data points.
+        targets: targets corresponding to instances in `data`.
+            Should have same length as `data`.
+        lookback: length of history to consider for prediction.
```

Comment:
I think "length" would fit better than "lookback" (length of timeseries samples)

Reply:
SGTM.